src/jshint.js

```diff
@@ -1242,6 +1242,30 @@ var JSHINT = (function() {
     return false;
   }
 
+  function isGlobalEval(left, state, funct) {
+    var isGlobal = false;
+
+    // permit methods to refer to an "eval" key in their own context
+    if (left.type === "this") {
```

Comment:
These two `if` statements should be merged and the following `if` changed to an `else if` (we want to avoid creating unnecessary code paths where possible).